### PR TITLE
feature(*): add new component creation guide for iOS

### PR DESCRIPTION
**Pull request by snooky23**
## Description
- add new component creation and debug guide for iOS

## Checklist

* [X] PR is scoped to one task
* [ ] Tests are included
* [ ] Documentation is included

* [ ] This PR is not making any UI change
* [ ] This PR is making a UI change
  * [ ] Screenshot(s) included

## QA :

* required test cases:
* specific apps / plugins to test :

### CODE REVIEW

File: book/ui-components/ios/ui-component-ios.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
+# UI Component
+
+## How to create new component
+
+1. In `ZLDefines.swift` add the new type of the component under `ZLComponentTypes` enum.
```
Comment (liviur) [inline book/ui-components/ios/ui-component-ios.md]: Where is this file , what repository , who can change it

File: book/ui-components/ios/ui-component-ios.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+2. In all the func in the `ZLComponentTypes` enum add  the following helper methods for the new component:
+    * `func typeByKey(_ key:String) -> ZLComponentTypes `
+    * `func componentType() -> String`
+    * `func builderClass() -> ZLComponentBuilderStructureProtocol.Type?`
+
+    Note: This is important because other classes will get the new component by key or use the relevant helpers.
```
Comment (liviur) [inline book/ui-components/ios/ui-component-ios.md]: set note inside italics by wrapping with `*`

File: book/ui-components/ui-component-general.md [hunk 3]
```diff
@@ -0,0 +1,8 @@
+# UI Component
+
+In this guide we'll focus on how to add a new component to the applicaster platform.
+
+For more information check the following guides:
+* Android - will be add soon
```
Comment (liviur) [inline book/ui-components/ui-component-general.md]: add -> added

File: book/ui-components/ios/ui-component-ios.md (hunk 1 above)
Comment (snooky23) [reply book/ui-components/ios/ui-component-ios.md]: Done.
I added the relevant information to the top of the section.

File: book/ui-components/ios/ui-component-ios.md (hunk 2 above)
Comment (snooky23) [reply book/ui-components/ios/ui-component-ios.md]: Done.

File: book/ui-components/ui-component-general.md (hunk 3 above)
Comment (snooky23) [reply book/ui-components/ui-component-general.md]: Done.

**Pull request merged**